GrpcClient/EmployeeGrpcRepository.cs

```diff
@@ -0,0 +1,286 @@
+            var reply = await client.GetSingleEmployeeByWorkingNumberAsync(new WorkingNumber { WorkingNumber_ = uint.CreateChecked(WorkingNumber)  }); 
+            employee = grpcEmployeeDtoToEntityEmployee(reply, GrpcShiftDtosToEntityShiftsList(reply.AssignedShifts.Dtos.ToList()));
+        }
+        catch (RpcException e)
+        {
+            if (e.StatusCode == StatusCode.NotFound)
+            {
+                throw new ArgumentException(e.Message);
+            }
+        }
+
+        return employee;
+    }
+
+    public async Task<bool> IsEmployeeInRepository(long Id)
+    {
+        System.Boolean b = new System.Boolean();
+        
+        using var channel = GrpcChannel.ForAddress(_grpcAddress); 
+        var client = new Employee.EmployeeClient(channel);
+        var reply = await client.IsEmployeeInRepositoryAsync(new Id { Id_ = Id });
+        b = reply.Boolean_;
+        
+        return b;
+    }
+
+    public static List<Entities.Shift> GrpcShiftDtosToEntityShiftsList(List<ShiftDTO> grpcShiftDtos)
+    {
+        List<Entities.Shift> shifts = new List<Entities.Shift>();
+        foreach (var shiftDTO in grpcShiftDtos)
+        {
+            var shift = new Entities.Shift
+            {
+                Id = shiftDTO.Id,
+                StartDateTime = new DateTime(shiftDTO.StartDateTime),
+                EndDateTime = new DateTime(shiftDTO.EndDateTime),
+                TypeOfShift = shiftDTO.TypeOfShift,
+                ShiftStatus = shiftDTO.ShiftStatus,
+                Description = shiftDTO.Description,
+                Location = shiftDTO.Location,
+                AssingnedEmployees = shiftDTO.AssignedEmployeeIds.ToList()
+            };
+            shifts.Add(shift);
+        }
+
+        return shifts;
+    }
+    
+    public static List<Entities.Shift> EntityShiftDtosToEntityShiftsList(List<DTOs.Shift.ShiftDTO> shiftDtos)
+    {
+        List<Entities.Shift> shifts = new List<Entities.Shift>();
+        foreach (var shiftDto in shiftDtos)
+        {
+            var shift = new Entities.Shift
+            {
+                Id = shiftDto.Id,
+                StartDateTime = shiftDto.StartDateTime,
+                EndDateTime = shiftDto.EndDateTime,
+                TypeOfShift = shiftDto.TypeOfShift,
+                ShiftStatus = shiftDto.ShiftStatus,
+                Description = shiftDto.Description,
+                Location = shiftDto.Location,
+                AssingnedEmployees = shiftDto.AssignedEmployees
+            };
+            shifts.Add(shift);
+        }
+
+        return shifts;
+    }
+
+    public static Entities.Employee grpcEmployeeDtoToEntityEmployee(EmployeeDTO employeeDto, List<Entities.Shift> shifts)
+    {
+        Entities.Employee employee = new Entities.Employee
+        {
+            FirstName = employeeDto.FirstName,
+            LastName = employeeDto.LastName,
+            Id = employeeDto.Id,
+            WorkingNumber = int.CreateChecked(employeeDto.WorkingNumber),
```

Comment:
What is CreateChecked and is it there instead of just employeeDto.WorkingNumber?

Reply:
converting int64/uint32 or some weird int format from java grpc to csharp int format